pkg/charts/charts.go

```diff
@@ -0,0 +1,299 @@
+	for i, chart := range charts {
+		if chart.Name == "admin-console" {
+			// admin-console has "embeddedClusterID" and "isAirgap" as dynamic values
+			newVals, err := setHelmValue(chart.Values, "embeddedClusterID", in.Spec.ClusterID)
+			if err != nil {
+				return nil, fmt.Errorf("set helm values admin-console.embeddedClusterID: %w", err)
+			}
+
+			newVals, err = setHelmValue(newVals, "isAirgap", fmt.Sprintf("%t", in.Spec.AirGap))
+			if err != nil {
+				return nil, fmt.Errorf("set helm values admin-console.isAirgap: %w", err)
+			}
+
+			newVals, err = setHelmValue(newVals, "isHA", in.Spec.HighAvailability)
+			if err != nil {
+				return nil, fmt.Errorf("set helm values admin-console.isHA: %w", err)
+			}
+
+			if in.Spec.Proxy != nil {
+				extraEnv := getExtraEnvFromProxy(in.Spec.Proxy.HTTPProxy, in.Spec.Proxy.HTTPSProxy, in.Spec.Proxy.NoProxy)
+				newVals, err = setHelmValue(newVals, "extraEnv", extraEnv)
+				if err != nil {
+					return nil, fmt.Errorf("set helm values admin-console.extraEnv: %w", err)
+				}
+			}
+
+			charts[i].Values = newVals
+		}
+		if chart.Name == "embedded-cluster-operator" {
+			// embedded-cluster-operator has "embeddedBinaryName" and "embeddedClusterID" as dynamic values
+			newVals, err := setHelmValue(chart.Values, "embeddedBinaryName", in.Spec.BinaryName)
+			if err != nil {
+				return nil, fmt.Errorf("set helm values embedded-cluster-operator.embeddedBinaryName: %w", err)
+			}
+
+			newVals, err = setHelmValue(newVals, "embeddedClusterID", in.Spec.ClusterID)
+			if err != nil {
+				return nil, fmt.Errorf("set helm values embedded-cluster-operator.embeddedClusterID: %w", err)
+			}
+
+			if in.Spec.Proxy != nil {
+				extraEnv := getExtraEnvFromProxy(in.Spec.Proxy.HTTPProxy, in.Spec.Proxy.HTTPSProxy, in.Spec.Proxy.NoProxy)
+				newVals, err = setHelmValue(newVals, "extraEnv", extraEnv)
+				if err != nil {
+					return nil, fmt.Errorf("set helm values embedded-cluster-operator.extraEnv: %w", err)
+				}
+			}
+
+			charts[i].Values = newVals
+		}
+		if chart.Name == "docker-registry" {
+			if !in.Spec.AirGap || !in.Spec.HighAvailability {
+				continue
+			}
+
+			serviceCIDR := util.ClusterServiceCIDR(*clusterConfig, in)
+			registryEndpoint, err := registry.GetRegistryServiceIP(serviceCIDR)
+			if err != nil {
+				return nil, fmt.Errorf("get registry service IP: %w", err)
+			}
+
+			newVals, err := setHelmValue(chart.Values, "service.clusterIP", registryEndpoint)
+			if err != nil {
+				return nil, fmt.Errorf("set helm values docker-registry.service.clusterIP: %w", err)
+			}
+
+			seaweedfsS3Endpoint, err := registry.GetSeaweedfsS3Endpoint(serviceCIDR)
+			if err != nil {
+				return nil, fmt.Errorf("get seaweedfs s3 endpoint: %w", err)
+			}
+
+			newVals, err = setHelmValue(newVals, "s3.regionEndpoint", seaweedfsS3Endpoint)
+			if err != nil {
+				return nil, fmt.Errorf("set helm values docker-registry.s3.regionEndpoint: %w", err)
+			}
+
+			charts[i].Values = newVals
```

Comment:
This is going to need to incorporate the changes from https://github.com/replicatedhq/embedded-cluster-operator/pull/240

```suggestion
		if chart.Name == "docker-registry" {
			if !in.Spec.AirGap {
				continue
			}
			
			// handle the registry IP, which will always be present in airgap
			serviceCIDR := util.ClusterServiceCIDR(*clusterConfig, in)
			registryEndpoint, err := registry.GetRegistryServiceIP(serviceCIDR)
			if err != nil {
				return nil, fmt.Errorf("get registry service IP: %w", err)
			}

			newVals, err := setHelmValue(chart.Values, "service.clusterIP", registryEndpoint)
			if err != nil {
				return nil, fmt.Errorf("set helm values docker-registry.service.clusterIP: %w", err)
			}
			charts[i].Values = newVals
			
			if !in.Spec.HighAvailability {
				continue
			}

			// handle the seaweedFS endpoint, which will only be present in HA airgap
			seaweedfsS3Endpoint, err := registry.GetSeaweedfsS3Endpoint(serviceCIDR)
			if err != nil {
				return nil, fmt.Errorf("get seaweedfs s3 endpoint: %w", err)
			}

			newVals, err = setHelmValue(newVals, "s3.regionEndpoint", seaweedfsS3Endpoint)
			if err != nil {
				return nil, fmt.Errorf("set helm values docker-registry.s3.regionEndpoint: %w", err)
			}

			charts[i].Values = newVals
```